ludwig/data/dataset/ray.py

```diff
@@ -64,14 +64,20 @@ def read_remote_parquet(path: str):
 
 
 class RayDataset(Dataset):
-    """Wrapper around ray.data.Dataset."""
+    """Wrapper around ray.data.Dataset.
+
+    Attributes:
+        auto_window: If True and the dataset is larger than available memory,
+            automatically set window size to `<available memory> // 5`.
+    """
 
     def __init__(
         self,
         df: Union[str, DataFrame],
         features: Dict[str, Dict],
         training_set_metadata: Dict[str, Any],
         backend: Backend,
+        auto_window: bool = True,
```

Comment:
@jeffkinnison for now, we can set the default value of `auto_window` to False.

In order to set `auto_window` to True, we'd need to do some benchmarking in order to figure out the impact of windowing. The `5 * dataset_size` heuristic is useful for determining the object store memory in order ensure that no disk spillage occurs. However, it's possible that even if the object store memory < `5 * dataset_size`, windowing is still expensive and doesn't lead to overall latency gains. In practice, the Anyscale team had mentioned that this is somewhere between `2 * dataset_size` and `3 * dataset_size`. So determining what that threshold looks like would be necessary before setting `auto_window` to True.

Separately, because of the new engines workstream, I expect that we would set `auto_window` to True if we're trying to cut down on resource requirements for an engine, in which case we would also need to have benchmarking. However, priority-wise, that would be a P1 for the new engines workstream that would be good to work on once the initial blockers for engines are resolved.

Reply:
This should be addressed in [dc3f08e](https://github.com/ludwig-ai/ludwig/pull/2721/commits/dc3f08ea0ab0f3aa39d8fcdcbde20b27c5b8c835). I agree that we should probably do an auto-sizing investigation at some point and see if we can come up with better window size heuristics.